server/src/config.rs

```diff
@@ -0,0 +1,137 @@
+// configuration framework rewrite time. very epic
+
+use directories::ProjectDirs;
+use serde::{Deserialize, Serialize};
+use std::fs::{self, File};
+use std::io::ErrorKind;
+use std::path::PathBuf;
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct Config {
+    #[serde(default = "max_password_len")]
+    pub max_password_len: u16,
+    #[serde(default = "min_password_len")]
+    pub min_password_len: u16,
+    #[serde(default = "max_username_len")]
+    pub max_username_len: u16,
+    #[serde(default = "min_username_len")]
+    pub min_username_len: u16,
+    #[serde(default = "max_display_name_len")]
+    pub max_display_name_len: u16,
+    #[serde(default = "min_display_name_len")]
+    pub min_display_name_len: u16,
+    #[serde(default = "profile_pictures")]
+    pub profile_pictures: PathBuf,
+    #[serde(default = "tokens_sweep_interval_secs")]
+    pub tokens_sweep_interval_secs: u64,
+    #[serde(default = "token_stale_days")]
+    pub token_stale_days: u16,
+    #[serde(default = "token_expiry_days")]
+    pub token_expiry_days: u16,
+}
+
+fn max_password_len() -> u16 {
+    1000
```

Comment:
not a review but hells yeah

Reply:
This is just a limit to stop people from being able to DOS us by sending too many characters and expecting us to hash it, lol. All passwords are the same size after hashing.

Reply:
very epic